llvm/lib/IR/RuntimeLibcalls.cpp

```diff
@@ -173,7 +173,7 @@ void RuntimeLibcallsInfo::initLibcalls(const Triple &TT) {
     setLibcallName(RTLIB::EXP10_F64, "__exp10");
   }
 
-  if (TT.isGNUEnvironment() || TT.isOSFuchsia() ||
+  if (TT.isGNUEnvironment() || TT.isOSFuchsia() || TT.isAmazonLinux() ||
```

Comment:
Adding a TT.isAmazonLinux() condition doesn't scale as there are many isGNUEnvironment conditions.

Ideally you should switch to `aarch64-amazon-linux-gnu`. As an alternative (but it's not a good idea to not switch to `linux-gnu`, set `Environment` to GNU, perhaps in `Triple::Triple`.

Reply:
> Ideally you should switch to `aarch64-amazon-linux-gnu`.

I agree, but I don't think that's a change I can make. AFAIK this is just the triple Amazon Linux uses. 

> As an alternative (but it's not a good idea to not switch to `linux-gnu`, set `Environment` to GNU, perhaps in `Triple::Triple`.

```
/// At its core the Triple class is designed to be a wrapper for a triple
/// string; the constructor does not change or normalize the triple string.
```
That would seem to go against the documentation for `Triple`.

Reply:
It looks like setting in the environment within `Triple::normalize` could work though.

Reply:
@MaskRay any thoughts on the above, or the possible alternative solution with `Triple::normalize` in https://github.com/llvm/llvm-project/pull/140070?

Reply:
Thanks for trying multiple solutions.  #140070 looks cleaner and avoids adding isAmazonlinux to other places isGNUEnvironment is used. Will we treat aarch64-amazon-linux-gnu as the triple and prefer it elsewhere? Perhaps yes. 

So we just need a way to recognize the aarch64-amazon-linux gcc installation.

Reply:
> So we just need a way to recognize the aarch64-amazon-linux gcc installation.

@MaskRay that's an interesting proposal worth pursuing. Do you have any specific mechanism of doing so in mind? At what level such recognition activity could be implemented?

Reply:
To restore LLVM's consistency, LLVM internally should use aarch64-amazon-linux-gnu. The configure-time LLVM_DEFAULT_TARGET_TRIPLE, and the target triple in clang --version output, in the runtime  library directory (lib/target/$triple), and the clang -S -emit-llvm output, should all use linux-gnu.

Perhaps we need a llvm/cmake/config.guess change.

clang driver needs a way to pick up aarch64-amazon-linux GCC installation. The user should specify `--gcc-triple=` correctly, or perhaps we can have a small hack within clang driver to do it.